query-service-impl/src/main/java/org/hypertrace/core/query/service/QueryServiceConfig.java

```diff
@@ -0,0 +1,117 @@
+package org.hypertrace.core.query.service;
+
+import com.typesafe.config.Config;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class QueryServiceConfig {
+
+  private static final String CONFIG_PATH_HANDLER_CLIENT_LIST = "clients";
+  private static final String CONFIG_PATH_HANDLER_CONFIG_LIST = "queryRequestHandlersConfig";
+  private static final String CONFIG_PATH_ATTRIBUTE_CLIENT = "attribute.client";
+
+  private final List<RequestHandlerClientConfig> requestHandlerClientConfigs;
+  private final List<RequestHandlerConfig> queryRequestHandlersConfigs;
+  private final ClientHostPortConfig attributeClientConfig;
+
+  QueryServiceConfig(Config config) {
+    Config resolved = config.resolve();
+    this.attributeClientConfig =
+        new ClientHostPortConfig(resolved.getConfig(CONFIG_PATH_ATTRIBUTE_CLIENT));
+    this.requestHandlerClientConfigs =
+        resolved.getConfigList(CONFIG_PATH_HANDLER_CLIENT_LIST).stream()
+            .map(RequestHandlerClientConfig::new)
+            .collect(Collectors.toUnmodifiableList());
+    this.queryRequestHandlersConfigs =
+        resolved.getConfigList(CONFIG_PATH_HANDLER_CONFIG_LIST).stream()
+            .map(RequestHandlerConfig::new)
+            .collect(Collectors.toUnmodifiableList());
+  }
+
+  public List<RequestHandlerClientConfig> getRequestHandlerClientConfigs() {
+    return this.requestHandlerClientConfigs;
+  }
+
+  public List<RequestHandlerConfig> getQueryRequestHandlersConfigs() {
+    return this.queryRequestHandlersConfigs;
+  }
+
+  public ClientHostPortConfig getAttributeClientConfig() {
+    return this.attributeClientConfig;
+  }
+
+  public static class RequestHandlerConfig {
+    private static final String CONFIG_PATH_NAME = "name";
+    private static final String CONFIG_PATH_TYPE = "type";
+    private static final String CONFIG_PATH_CLIENT_KEY = "clientConfig";
+    private static final String CONFIG_PATH_REQUEST_HANDLER_INFO = "requestHandlerInfo";
+
+    private final String name;
+    private final String type;
+    private final String clientConfig;
+    private final Config requestHandlerInfo;
+
+    private RequestHandlerConfig(Config config) {
+      this.name = config.getString(CONFIG_PATH_NAME);
```

Comment:
Just for my own understanding, what's the difference b/w this model and the previous one like:
`ConfigBeanFactory.create(config, QueryServiceImplConfig.class);`?

Reply:
Discussed in the PR description - https://github.com/hypertrace/query-service/pull/23#issue-491929367 
Specifically, the problem that drove this was the fact that every top level property in the bean object had to correspond to a top level config property.